addon/components/frost-modal-input.js

```diff
@@ -1,9 +1,129 @@
+      return
+    }
+
+    this.set('containerObserver', mutationObserver)
+  },
+
+  /**
+  * Checks scroll position within container element to add/remove scroll styling to header/footer elements
+  */
+  updateScrollStyles () {
+    const $containerEl = this.get('$containerEl')
+    const $headerEl = this.get('$headerEl')
+    const $footerEl = this.get('$footerEl')
+    const scrollTop = $containerEl.scrollTop()
+    const innerHeight = $containerEl.innerHeight()
+    const scrollHeight = $containerEl.get(0) && $containerEl.get(0).scrollHeight
+    this.updatePerfectScrollbar()
+
+    // style top of form with box shadow if content overflow hidden underneath header
+    const shouldHeaderClassBePresent = scrollTop > 0
+    $headerEl.toggleClass('header-scrolled', shouldHeaderClassBePresent)
+
+    // style bottom of form with box shadow if content overflow hidden underneath footer
+    const shouldFooterClassBePresent = scrollTop + innerHeight < scrollHeight
+    $footerEl.toggleClass('footer-scrolled', shouldFooterClassBePresent)
+  },
+
+  /**
+   * sets bindings/callbacks for perfect-scrollbar events on document element
+   * creates a window.MutationObserver for mutating events on container element
+   */
+  setScrollBindings () {
+    const $headerEl = this.get('$headerEl')
+    const $footerEl = this.get('$footerEl')
+
+    this.set('containerObserver', this.createMutationObserver())
+
+    // bind document to perfect-scrollbar events
+    Ember.$(document).on('ps-scroll-up', () => $footerEl.addClass('footer-scrolled'))
+    Ember.$(document).on('ps-y-reach-end', () => $footerEl.removeClass('footer-scrolled'))
+    Ember.$(document).on('ps-scroll-down', () => $headerEl.addClass('header-scrolled'))
+    Ember.$(document).on('ps-y-reach-start', () => $headerEl.removeClass('header-scrolled'))
+
+    Ember.$(window).resize(this.updateScrollStyles.bind(this))
+
+    // update scroll styles in case where content already hidden on initial DOM render
+    this.updateScrollStyles()
+
+    this.set('scrollBindingsSet', true)
+  },
+
+  actions: {
+    modalOpen () {
+      this.set('$containerEl', Ember.$('.ember-remodal.window.remodal-is-opened .ps-container'))
+      this.set('$headerEl', Ember.$('.ember-remodal.window.remodal-is-opened .input-header'))
+      this.set('$footerEl', Ember.$('.ember-remodal.window.remodal-is-opened .actions'))
+
+      if (!(this.get('$containerEl').length > 0 && this.get('$headerEl').length > 0 && this.get('$footerEl').length > 0)) {
+        return
+      }
+      if (!this.get('scrollBindingsSet')) {
+        this.setScrollBindings()
+      }
+    },
+    modalClose () {
+      Ember.$(document).off('ps-scroll-up ps-scroll-down ps-y-reach-start ps-y-reach-end')
+
+      Ember.$(window).off('resize', this.updateScrollStyles)
+
+      if (this.get('containerObserver')) {
+        this.get('containerObserver').disconnect()
+      }
+      this.set('scrollBindingsSet', false)
+    }
+  }
```

Comment:
Cool...you moved the scroll bindings attach/remove to modalOpen/modalClose.  I was worried about my overlooking calling this._super() in willDestroyElement() but you have removed the need for this.  Nice refactor:)